### PR TITLE
fixing progress counter bug

### LINKED ISSUE #196: progress counts are inaccurate
In following example reported in the logs by a single thread:
```
23/08/30 13:48:31 INFO DiffJobSession: ThreadID: 109 Read Record Count: 52600000
23/08/30 13:48:31 INFO DiffJobSession: ThreadID: 109 Mismatch Record Count: 4
23/08/30 13:48:31 INFO DiffJobSession: ThreadID: 109 Corrected Mismatch Record Count: 4
23/08/30 13:48:31 INFO DiffJobSession: ThreadID: 109 Missing Record Count: 0
23/08/30 13:48:31 INFO DiffJobSession: ThreadID: 109 Corrected Missing Record Count: 0
23/08/30 13:48:31 INFO DiffJobSession: ThreadID: 109 Valid Record Count: 52587843
23/08/30 13:48:31 INFO DiffJobSession: ThreadID: 109 Skipped Record Count: 0
```

The Valid + Corrected do not match the Read count.

At issue is the threads are reporting global totals, and these totals are not updated as an atomic grouping.

### PR DESCRIPTION
**Pull request by mieslep**
1. Consolidates progress logging into helper class `JobCounter`
2. Separates counts within each thread from the global counts
3. Increments global counts as each thread completes
4. Introduces new parameter to enable per-part logging
5. Per-part logging on a single line per part

**Which issue(s) this PR fixes**:
Fixes #196

**Checklist:**
- [X] Automated Tests added/updated
- [X] Documentation added/updated
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: SIT/regression/03_performance/migrate.properties [hunk 1]
```diff
@@ -6,3 +6,6 @@ spark.cdm.schema.target.keyspaceTable             target.regression_performance
 
 spark.cdm.perfops.numParts                        32
 spark.cdm.perfops.batchSize                       1
+
+spark.cdm.perfops.printStatsAfter                 450
```
Comment (msmygit) [inline SIT/regression/03_performance/migrate.properties:10]: LGTM. Let's please coordinate with docs crew to publish this at https://docs.datastax.com/en/astra-serverless/docs/migrate/cassandra-data-migrator.html page

File: src/test/java/com/datastax/cdm/properties/PropertyHelperTest.java [hunk 2]
```diff
@@ -498,12 +498,12 @@ public void test_valid_connection_noUserPassword() {
 
     @Test
     public void meetsMinimum_true() {
-        assertTrue(helper.meetsMinimum("a", 100,0));
+        assertTrue(helper.meetsMinimum("a", 100L,0L));
```
Comment (msmygit) [inline src/test/java/com/datastax/cdm/properties/PropertyHelperTest.java:501]:
```suggestion
        assertTrue(helper.meetsMinimum("a", 100L, 0L));
```

File: src/test/java/com/datastax/cdm/properties/PropertyHelperTest.java [hunk 3]
```diff
@@ -498,12 +498,12 @@ public void test_valid_connection_noUserPassword() {
...
     }
 
     @Test
     public void meetsMinimum_false() {
-        assertFalse(helper.meetsMinimum("a", 1,100));
+        assertFalse(helper.meetsMinimum("a", 1L,100L));
```
Comment (msmygit) [inline src/test/java/com/datastax/cdm/properties/PropertyHelperTest.java:506]:
```suggestion
        assertFalse(helper.meetsMinimum("a", 1L, 100L));
```

File: src/main/java/com/datastax/cdm/job/JobCounter.java [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+        // Remove the trailing comma and space
+        if (sb.length() > 2) {
+            sb.setLength(sb.length() - 2);
+        }
+        return sb.toString();
+    }
+
+    public void printProgress() {
+        printProgress(false);
+        printProgress(true);
```
Comment (msmygit) [inline src/main/java/com/datastax/cdm/job/JobCounter.java:137]: I don't understand what we're doing in here by calling both `true` and `false`?

**Comment by msmygit** [comment]
Any idea why we're receiving [these `ERROR`s](https://github.com/datastax/cassandra-data-migrator/actions/runs/6073955848/job/16476859913?pr=197#step:4:1129) in the build output and how do we address them?

**Review by msmygit — APPROVED** [review]

**Review by pravinbhat — CHANGES_REQUESTED** [review]

**Comment by msmygit** [comment]
I've encorporated the review comments and pushed the changes. Merging here.

**Pull request merged**